tests/integration/v0x04/test_match.py

```diff
@@ -20,3 +20,17 @@ def test_dl_vlan_pyof(self):
         of_tlv = expected.as_of_tlv()
         actual = MatchDLVLAN.from_of_tlv(of_tlv)
         self.assertEqual(expected, actual)
+
+    def test_dl_vlan_mask_pyof(self):
```

Comment:
Great test cases. Now that we have both serialization and deserialization resulting in the same expected `MatchDLVLAN` consistency check is supposed to work correctly as well. 

If you have collected some bytes on wire, and if you wish, you an also add some deserialization unit tests ensuring that something like `test_dl_vlan_untagged_from_ovs`. 

Finally, make sure to plan for e2e tests ensuring that these flows can be installed on OvS (which shouldn't be much a surprise when running the tests since you've been extensively exploring this with OvS as you're developing but then we have these e2e tests as regression tests too)

Reply:
Which match field handles the bytes on wire?

Reply:
@Alopalao, at the lowest level, it's handled on [`on_raw_in`](https://github.com/kytos-ng/of_core/blob/master/main.py#L260-L271) from bytes, slicing, and deserializing with `unpack(bytes)`.

To simplify, mocking and capturing the bytes we're interested in, you could assume that we're already deserializing correct (which we are), and then on `_handle_multipart_flow_stats`, you could add these temporary log lines:

```
diff --git a/main.py b/main.py
index dd14e58..de3990a 100644
--- a/main.py
+++ b/main.py
@@ -176,6 +176,9 @@ class Main(KytosNApp):
 
         Returns true if no more replies are expected.
         """
+        log.info(f"_handle_multipart_flow_stats reply: {reply}, body: {reply.body}")
+        for of_flow_stats in reply.body:
+            log.info(f"of_flow_stats {of_flow_stats}, body packed: {of_flow_stats.pack()}")
         if self._is_multipart_reply_ours(reply, switch, 'flows'):
             # Get all flows from the reply and extend the multipar flows list
             flows = [Flow04.from_of_flow_stats(of_flow_stats, switch)
@@ -299,6 +302,7 @@ class Main(KytosNApp):
                     continue
 

```

That way you'd have serialized bytes representations from `pyof.v0x04.controller2switch.multipart_reply.FlowStats`, you could capture these bytes, then you can confirm on wireshark that they're the same (since we're packing here again to facilitate we're already using your code calling `as_of_tlv` - so confirming on wireshark we remove the bias of the code introduced here), and then use them in unit tests instantiating `FlowStats` calling `unpack(bytes)` and then ensuring that it results in the same expected `match`. Here's some random `FlowStats` that I currently have (not related to untagged or any though):

```
2023-03-06 15:53:43,842 - INFO [kytos.napps.kytos/of_core] (MainThread) of_flow_stats <pyof.v0x04.controller2switch.multipart_reply.FlowStats object at 0x7ff34c797a30>, body packed: b'\
x00X\x00\x00\x00\x00\x00\x051\r\xfb\xc0\xc3P\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\xac\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\
x01\x00\x0e\x80\x00\x08\x06\xee\xee\xee\xee\xee\x01\x00\x00\x00\x04\x00\x18\x00\x00\x00\x00\x00\x00\x00\x10\xff\xff\xff\xfd\xff\xff\x00\x00\x00\x00\x00\x00'
2023-03-06 15:53:43,842 - INFO [kytos.napps.kytos/of_core] (MainThread) of_flow_stats <pyof.v0x04.controller2switch.multipart_reply.FlowStats object at 0x7ff34c77edf0>, body packed: b'\
x00X\x00\x00\x00\x00\x00\x050\xfe\xb9\x80\xc3P\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\xac\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0
0\x01\x00\x0e\x80\x00\x08\x06\xee\xee\xee\xee\xee\x02\x00\x00\x00\x04\x00\x18\x00\x00\x00\x00\x00\x00\x00\x10\xff\xff\xff\xfd\xff\xff\x00\x00\x00\x00\x00\x00'
2023-03-06 15:53:43,843 - INFO [kytos.napps.kytos/of_core] (MainThread) of_flow_stats <pyof.v0x04.controller2switch.multipart_reply.FlowStats object at 0x7ff34c77ef40>, body packed: b'\
x00X\x00\x00\x00\x00\x00\x051\xa6\x92@\x03\xe8\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00\xab\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x00\x00\x00\x10\x86\x00\x00\x00\x00\x00\x02\xb5\xfc\x0
0\x01\x00\x10\x80\x00\n\x02\x88\xcc\x80\x00\x0c\x02\x1e\xd7\x00\x04\x00\x18\x00\x00\x00\x00\x00\x00\x00\x10\xff\xff\xff\xfd\xff\xff\x00\x00\x00\x00\x00\x00'
```

Reply:
I think I am going to explore this later.